src/interactivity/base/InteractivityFactory.cpp

```diff
@@ -29,6 +29,88 @@
 using namespace std;
 using namespace Microsoft::Console::Interactivity;
 
+HRESULT PseudoConsoleWindowAccessibilityProvider::RuntimeClassInitialize(HWND pseudoConsoleHwnd) noexcept
+{
+    RETURN_HR_IF_NULL(E_INVALIDARG, pseudoConsoleHwnd);
+    _pseudoConsoleHwnd = pseudoConsoleHwnd;
+    return S_OK;
+}
+
+IFACEMETHODIMP PseudoConsoleWindowAccessibilityProvider::get_ProviderOptions(_Out_ ProviderOptions* pOptions)
+{
+    RETURN_HR_IF_NULL(E_INVALIDARG, pOptions);
+    *pOptions = ProviderOptions_ServerSideProvider;
+    return S_OK;
+}
+
+IFACEMETHODIMP PseudoConsoleWindowAccessibilityProvider::GetPatternProvider(_In_ PATTERNID /*iid*/,
+                                                                            _COM_Outptr_result_maybenull_ IUnknown** ppInterface)
+{
+    RETURN_HR_IF_NULL(E_INVALIDARG, ppInterface);
+    *ppInterface = nullptr;
+    return S_OK;
+}
+
+IFACEMETHODIMP PseudoConsoleWindowAccessibilityProvider::GetPropertyValue(_In_ PROPERTYID propertyId,
+                                                                          _Out_ VARIANT* pVariant)
+{
+    RETURN_HR_IF_NULL(E_INVALIDARG, pVariant);
+
+    pVariant->vt = VT_EMPTY;
+
+    // Returning the default will leave the property as the default
+    // so we only really need to touch it for the properties we want to implement
+    if (propertyId == UIA_ControlTypePropertyId)
+    {
+        pVariant->vt = VT_I4;
+        pVariant->lVal = UIA_WindowControlTypeId;
+    }
+    else if (propertyId == UIA_NamePropertyId)
+    {
+        pVariant->bstrVal = SysAllocString(AutomationPropertyName);
+        if (pVariant->bstrVal != nullptr)
+        {
+            pVariant->vt = VT_BSTR;
+        }
+    }
+    else if (propertyId == UIA_IsControlElementPropertyId)
+    {
+        pVariant->vt = VT_BOOL;
+        pVariant->boolVal = VARIANT_FALSE;
+    }
+    else if (propertyId == UIA_IsContentElementPropertyId)
+    {
+        pVariant->vt = VT_BOOL;
+        pVariant->boolVal = VARIANT_FALSE;
+    }

```

Comment:
@DHowett @zadjii-msft 
So, in XAML, you can set the `AccessibilityView` to `Raw` to remove it from the UIA tree. This is actually an abstraction:
| AccessibilityView| IsControlElement | IsContentElement |
|--|--|--|
| Raw | false | false |
| Control | true | false |
| Content| true | true |

Source: https://learn.microsoft.com/en-us/windows/win32/winauto/uiauto-treeoverview

By setting both of these to false, we're essentially "removing" this from the UIA tree. _However_, you can't _really_ remove it. It's just accessible if you want the raw version of the tree.

This distinction is particularly important for Accessibility Insights. Unfortunately, the FastPass tool navigates through the UIA tree and looks for common errors. So, it still finds this PseudoConsoleWindow and complains that it needs a name (unfortunately).

To make matters worse, even with this fix where we change it to `Raw` and give it a name, we're still getting a bug from FastPass saying that we _should_ set `IsControlElement` to true. But for the reasons you've stated, we obviously don't want that.

Idk if there's a way to exclude this window from FastPass. Like we said, it's not focusable. We basically want to completely omit this from the UIA tree, but this is as close as we can get, it seems.

Reply:
_Got it!_ Thank you!